### PR TITLE
VB-3901 - Local visitor restriction change events are processed

### PR DESCRIPTION
**Pull request by adaviesMOJ**
## What does this pull request do?

When the visit-scheduler is alerted through the notifications controller of a change in a visitors (person) restriction, flag any visits the person has for a specific prisoner.

## What is the intent behind these changes?

Enable tracking of prison-offender-events.prisoner.person-restriction.changed events.

## JIRA Internal Link
https://dsdmoj.atlassian.net/browse/VB-3901

### CODE REVIEW

**Review by dhirajshettymoj — APPROVED** [review]
LGTM

File: src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/dto/visitnotification/PersonRestrictionDeletedNotificationDto.kt [hunk 1]
```diff
@@ -15,4 +15,6 @@ data class PersonRestrictionChangeNotificationDto(
   val validFromDate: LocalDate,
   @JsonInclude(Include.NON_NULL)
   val validToDate: LocalDate? = null,
```
[the remark this answers is not in the document]
Comment (adaviesMOJ) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/dto/visitnotification/PersonRestrictionDeletedNotificationDto.kt:17]: As mentioned on slack, they are sent as the deleted event structure (including additional information) is identical to the original changed event structure.

File: src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/dto/visitnotification/PersonRestrictionUpsertedNotificationDto.kt [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+import com.fasterxml.jackson.annotation.JsonInclude.Include
+import jakarta.validation.constraints.NotBlank
+import jakarta.validation.constraints.NotNull
+import java.time.LocalDate
+
+data class PersonRestrictionUpsertedNotificationDto(
+  @NotBlank
+  val prisonerNumber: String,
+  @NotBlank
+  val visitorId: String,
```
[the remark this answers is not in the document]
Comment (adaviesMOJ) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/dto/visitnotification/PersonRestrictionUpsertedNotificationDto.kt:13]: Will implement this, thank you.

**Review by dhirajshettymoj — APPROVED** [review]
LGTM

**Pull request merged**